airflow/sensors/external_task_sensor.py

```diff
@@ -136,6 +140,7 @@ def poke(self, context, session=None):
                     raise AirflowException('The external task'
                                            '{} in DAG {} does not exist.'.format(self.external_task_id,
                                                                                  self.external_dag_id))
+        self.has_check_existence = True
```

Comment:
Maybe this line can be put inside the if-clause? Logically more meaningful.

Reply:
update